source/developer/mobile-developer-setup.rst

```diff
@@ -117,10 +188,128 @@ Android (Device)
+
+You will always be able to build an unsigned version of the app as it does not need provisioning profiles or certificates as long as you set up Fastlane in your environment.
+
+ - **make build-ios**: Builds the iOS app and generates the ``.ipa`` file to be distributed. This make command expects an argument as the target which can be ``dev``, ``beta`` or ``release``. Depending on the target, a Fastlane script runs and each lane has the appropriate certificates and steps according to the Mattermost release process.
+ - **make build-android**: Builds the Android app and generates the ``.apk`` file to be distributed. This make command expects an argument as the target which can be ``dev``, ``alpha`` or ``release``. Depending on the target, a Fastlane script runs and each lane has the appropriate certificates and steps according to the Mattermost release process.
+ - **make unsigned-ios**: Builds the iOS app and generates an unsigned ``Mattermost-unsigned.ipa`` file in the project's root directory.
+ - **make unsigned-android**: Builds the Android app and generates an unsigned ``Mattermost-unsigned.apk`` file in the project's root directory.
+
+If you plan to use the **make build-*** commands, be sure to `modify Fastlane <https://docs.fastlane.tools/>`_ to suit your needs or the commands will fail.
+
+Adding New Dependencies to the Project
+-------------------------------------------
+
+If you need to add a new dependency to the project, it is important to add them in the right way. Instructions for adding different types of dependencies are described below:
+
+JavaScript Only
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you need to add a new JavaScript dependency that is not related to React Native, **use yarn, not npm**. Be sure to save the exact version number to avoid conflicts in the future. 
+
+	e.g. ``yarn add -E <package-name>``
+  
+React Native
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+As with ``JavaScript only``, **use yarn** to add your dependency and include an exact version. Then link the library in react native by running ``react-native link <package-name>`` in a terminal.
+
+Be aware that we are using React Native Navigation. For Android, you might need to complete the linking process manually as the react-native link command won't do it for you.
+
+After running the react-native link command, head to ``<project-root>/android/app/src/main/java/com/mattermost/rnbeta/MainApplication.java`` and initialize the react native library that you just added in the ``createAdditionalReactPackages`` method.
+
+Android
+~~~~~~~~~~~~
+
+Usually the react native libraries that you add to the project will take care of adding new dependancies to the project. 
+
+If you come across a case where it is needed, we recommend you first review your work to confirm. The Android documentation should then be followed to add the libraries.
```

Comment:
Is there a link to Android documentation we can give?

Reply:
No documentation cause that depends on each dependency that the developer may want to add